bull/pdl-queue/consumer.js

```diff
@@ -1,175 +1,221 @@
+    )
+      .text()
+      .trim();
+    PNdetails.description = PNdetails.description.replace(/\n/g, "");
+    PNdetails.description = PNdetails.description.replace(/\[edit]/g, "");
+  }
+
+  PNdetails.title = $(
+    "#Nanakshahi > tbody > tr:nth-child(3) > td > table > tbody > tr:nth-child(2) > td > table > tbody > tr > td:nth-child(2) > table:nth-child(22) > tbody > tr:nth-child(3) > td > table > tbody > tr:nth-child(2) > td > table > tbody > tr > td:nth-child(2) > div > table > tbody > tr:nth-child(1) > td > a"
+  )
+    .text()
+    .trim();
+
+  PNdetails.bookID = bookid;
+  PNdetails.categoryID = categoryID;
+
+  delete PNdetails[""];
+
+  return PNdetails;
 }
 
 async function getZipAndBytelength(no_of_pages, id, title) {
-    var zip = new JSZip();
-    title = title.replace(/ /g, "_")
-    var img = zip.folder(`${title}_images`);
-    let temp_pages = no_of_pages;
-    var download_image = async function (uri, filename) {
-        await rp({
-            method: "GET",
-            uri,
-            encoding: null,
-            transform: function (body, response) {
-                return { headers: response.headers, data: body };
-            }
-        })
-            .then(async function (body) {
-                if (/image/.test(body.headers["content-type"])) {
-                    var data = new Buffer(body.data)
-                    img.file(filename, data.toString("base64"), { base64: true });
-                }
-            })
-            .catch(function (err) {
-                --no_of_pages;
-            })
-    }
-    for (let i = 1; i <= temp_pages; ++i) {
-        const str = `http://www.panjabdigilib.org/images?ID=${id}&page=${i}&pagetype=null&Searched=W3GX`;
-        await download_image(str, `${title}_${i}.jpeg`)
-        console.log("Image " + i + " zipped.");
-    }
-    let { byteLength } = await zip.generateAsync({ type: 'nodebuffer' })
-    byteLength = Number(byteLength + no_of_pages * 16) //No. of pages * 16
-    return [zip, byteLength]
+  var zip = new JSZip();
+  title = title.replace(/ /g, "_");
+  var img = zip.folder(`${title}_images`);
+  let temp_pages = no_of_pages;
+  var download_image = async function(uri, filename) {
+    await rp({
+      method: "GET",
+      uri,
+      encoding: null,
+      transform: function(body, response) {
+        return { headers: response.headers, data: body };
+      }
+    })
+      .then(async function(body) {
+        if (/image/.test(body.headers["content-type"])) {
+          var data = new Buffer(body.data);
+          img.file(filename, data.toString("base64"), { base64: true });
+        }
+      })
+      .catch(function(err) {
+        --no_of_pages;
+      });
+  };
+  for (let i = 1; i <= temp_pages; ++i) {
+    const str = `http://www.punjabdigilib.org/images?ID=${id}&page=${i}&pagetype=null&Searched=W3GX`;
```

Comment:
The URLs are not supposed to be changed. The website itself is : http://www.panjabdigilib.org/webuser/searches/mainpage.jsp 
Changing the URL will give us a 404 Error.